### ISSUE TITLE
Custom reporter in Cypress config doesn't work in Browserstack

### ISSUE THREAD

**Issue by theVoogie** [browserstack/browserstack-cypress-cli#137]
I'm using Cypress to run my e2e testing on Browserstack. I'm using `junit` reporter which is configured in the cypress.json. When I'm running tests locally test results are generated according to the format (i.e junit). However, when I'm running testing in Browserstack and fetch the results, they are not following the cypress config and are returned as mocha report files (json/html) instead of junit (xml).

 **cypress.json**
```
{
  "reporter": "junit",
  "reporterOptions": {
    "mochaFile": "results/test-output-[hash].xml",
    "toConsole": true,
    "attachments": true
  }
}
```

**browserstack.json**
```
{
    "auth": {...},
    "browsers": [...],

    "run_settings": {
      "cypress_config_file": "./cypress.json",
      "cypress_version": "6",
      "project_name": "FancyProjectName",
      "build_name": "Build no. 42",
      "parallels": "2",
      "npm_dependencies": {
        "typescript": "^4.2.4"
      }
    }
  }
```

**Comment by HeenaNanwani8** [comment]
Hey @theVoogie, currently, there is no way to retrieve the reports generated on the remote machine.  On BrowserStack, you can generate the HTML report for your test builds. More details on https://www.browserstack.com/docs/automate/cypress/generate-reports.

**Comment by karanshah-browserstack** [comment]
Will be fixed by #151
